source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -717,20 +728,23 @@ module Error = struct
         labels: string list;
       }
     | InvalidMultiSink of string
-    | RuleCodeDuplicate of int
+    | RuleCodeDuplicate of {
+        code: int;
+        previous_location: JsonAst.LocationWithPath.t option;
+      }
     | OptionDuplicate of string
     | SourceDuplicate of string
     | SinkDuplicate of string
     | TransformDuplicate of string
     | FeatureDuplicate of string
-  [@@deriving equal]
+  [@@deriving equal, compare]
```

Comment:
Why do we need `compare` here (and below)?
```suggestion
  [@@deriving equal]
```

Reply:
So previously I used a List.`dedup and sort` to prevent raising two errors when a combined_source_rule and rule both contained the same code which required compare. Now, I migrated to use a Hashset which is more optimal for this comparison. We will need the comparison since a combined_source_rule can generate more than one rule..